File: java/src/sample/replicant/sample/MockRequestQueue.java
// Copyright 2009 Kiel Hodges
package replicant.sample;

import replicant.*;
import static replicant.Replicant.*;

public class MockRequestQueue implements RequestQueue {

  public Request nextRequest() {
    return nextRequest.call();
  }
  
  public final MethodMocker0<Request> nextRequest = 
    methodMocker(this, "nextRequest").orderingResponses().with0Args();
  
}

File: scala/src/test/scala/sample/MockRequestQueue.scala
// Copyright 2011 Kiel Hodges
package sample

import replicant._

class MockRequestQueue extends Mock[RequestQueue] { minder =>

  val mock: RequestQueue = new RequestQueue {
    def nextRequest: Option[Request] = minder.nextRequest()
  }

  val nextRequest = method("nextRequest", mock.nextRequest _)

}
